src/pages/home/sidebar/SidebarLinksData.tsx

```diff
@@ -259,50 +209,38 @@ const chatReportSelector = (report) =>
         isUnreadWithMention: ReportUtils.isUnreadWithMention(report),
     };
 
-/**
- * @param {Object} [reportActions]
- * @returns {Object|undefined}
- */
-const reportActionsSelector = (reportActions) =>
+const reportActionsSelector = (reportActions: OnyxEntry<OnyxTypes.ReportActions>) =>
     reportActions &&
-    lodashMap(reportActions, (reportAction) => {
-        const {reportActionID, parentReportActionID, actionName, errors = [], originalMessage} = reportAction;
-        const decision = lodashGet(reportAction, 'message[0].moderationDecision.decision');
+    Object.values(reportActions).map((reportAction) => {
+        const {reportActionID, actionName, errors, originalMessage} = reportAction;
+        const decision = reportAction.message?.[0].moderationDecision?.decision;
 
         return {
             reportActionID,
-            parentReportActionID,
             actionName,
             errors,
             message: [
                 {
                     moderationDecision: {decision},
-                },
+                } as Message,
             ],
             originalMessage,
         };
     });
 
-/**
- * @param {Object} [policy]
- * @returns {Object|undefined}
- */
-const policySelector = (policy) =>
+const policySelector = (policy: OnyxEntry<OnyxTypes.Policy>) =>
     policy && {
         type: policy.type,
         name: policy.name,
         avatar: policy.avatar,
     };
 
-export default compose(
-    withCurrentReportID,
-    withCurrentUserPersonalDetails,
-    withNavigationFocus,
-    withNetwork(),
-    withOnyx({
+export default withCurrentReportID(
+    withOnyx<SidebarLinksDataProps, SidebarLinksDataOnyxProps>({
         chatReports: {
             key: ONYXKEYS.COLLECTION.REPORT,
-            selector: chatReportSelector,
+            // This assertion is needed because the selector in withOnyx expects that the return type will be the same as type in ONYXKEYS but for collection keys the selector is executed for each collection item. This is a bug in withOnyx typings that we don't have a solution yet, when useOnyx hook is introduced it will be fixed.
```

Comment:
There was a recent PR in onyx, maybe it resolved the issue here?

Reply:
Yes @kubabutkiewicz The Onyx PR was merged and is available on `main`, let's remove the comments and apply that code I suggested to you

Reply:
will do that :)

Reply:
@blazejkustra @fabioh8010 its adjusted , you can recheck